internal/ignition/ignition.go

```diff
@@ -384,13 +384,21 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte,
 		log.Infof("adding manifest %s to working dir for cluster %s", manifest, g.cluster.ID)
 		err = g.downloadManifest(ctx, manifest)
 		if err != nil {
-			_ = os.Remove(filepath.Join(g.workDir, "manifests"))
-			_ = os.Remove(filepath.Join(g.workDir, "openshift"))
+			defer os.Remove(filepath.Join(g.workDir, "manifests"))
```

Comment:
`defer` doesn't do anything here

What I had in mind  was to run:

```go
defer func() {
	if err != nil {
		os.Remove(filepath.Join(g.workDir, "manifests"))
		os.Remove(filepath.Join(g.workDir, "openshift"))
	}
}
```

In the beginning of the function. This way if any error happens we get automatic cleanup in all error branches

Reply:
:facepalm:  my bad, now I see what you meant.